src/lib_ccx/file_functions.c

```diff
@@ -386,12 +390,35 @@ size_t buffered_read_opt (struct ccx_demuxer *ctx, unsigned char *buffer, size_t
 				int keep = ctx->bytesinbuffer > 8 ? 8 : ctx->bytesinbuffer;
 				memmove (ctx->filebuffer, ctx->filebuffer+(FILEBUFFERSIZE-keep),keep);
 				int i;
+				struct sockaddr_in source_addr;
+				socklen_t len = sizeof(source_addr);
+				/* Get address of host to check for udp network mutlicasting */
+				in_addr_t addr;
+				if (ccx_options.udpaddr != NULL)
+				{
+					struct hostent *host = gethostbyname(ccx_options.udpaddr);
+					addr = ntohl(((struct in_addr *)host->h_addr_list[0])->s_addr);
+				}
+				else
+				{
+					addr = INADDR_ANY;
+				}
+
 				if (ccx_options.input_source == CCX_DS_FILE || ccx_options.input_source == CCX_DS_STDIN)
 					i = read (ctx->infd, ctx->filebuffer + keep, FILEBUFFERSIZE-keep);
 				else if (ccx_options.input_source == CCX_DS_TCP)
 					i = net_tcp_read(ctx->infd, (char *) ctx->filebuffer + keep, FILEBUFFERSIZE - keep);
-				else
-					i = recvfrom(ctx->infd,(char *) ctx->filebuffer + keep, FILEBUFFERSIZE - keep, 0, NULL, NULL);
+				else {
```

Comment:
Create a function for udp data reading in file src/lib_ccx/networking.c similar to net_tcp_read.

Reply:
Ok. Sure.

Reply:
@anshul1912 Is it okay now ?